app/controllers/vm_common.rb

```diff
@@ -1677,6 +1677,39 @@ def set_form_vars
     session[:edit] = @edit
   end
 
+  ACTION_TYPES = {
+    "advanced_settings"   => ->(amount) { n_("Advanced Setting", "Advanced Settings", amount) },
```

Comment:
The ambiguity: If the problem appears, we add the namespace as ` ui_action|Groups`. At this point I don't see this as an issue. By the way the translators would have no idea what 'ui_actions' mean unless someone writes a document on such things.

Gettext approach and i18n yaml approach have both their plus and cons. This particular place might seem it fits better for the i18n interface, but it was modeled around that interface in the first place.

Dumping huge string maps: yes, in this particular place we are adding a string map to the code. We can put that in a separate module or something if it's a problem for you.

The strings are easier to maintain if they are closer to the place where they are actually used. And if they appear only once. Unlike if you have to maintain consistency between keys used and keys in the yaml file.

As @mzazrivec goes through the yaml file he often finds strings that are no longer used or strings those use is hard to find in the codebase.

But we don't want to get into gettext/i18n discussion, let's focus on finding the pattern that works in this context.

So I see 2 objections that where not addressed yet:

 1. lambdas capturing too much -- is that a problem? can we solve that by creating the hash in a function and assigning it to an instance variable for reuse (instead of a constant)?

 2. adding large map directly to the controller -- is that a problem? does this have to be solved right here and now (blocking the i18n work)? can we resolve that by moving the hash/function to a module?